File: subnets/src/main/java/com/clouway/subnets/core/Category.java
package com.clouway.subnets.core;

/**
 * @author Marian Zlatev <mzlatev91@gmail.com>
 */
public final class Category {

  public final String id;
  public final String type;

  public Category(String id, String type) {
    this.id = id;
    this.type = type;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof Category)) return false;

    Category category = (Category) o;

    if (id != null ? !id.equals(category.id) : category.id != null) return false;
    return !(type != null ? !type.equals(category.type) : category.type != null);

  }

  @Override
  public int hashCode() {
    int result = id != null ? id.hashCode() : 0;
    result = 31 * result + (type != null ? type.hashCode() : 0);
    return result;
  }
}

File: ngbp/src/app/modules/category.js
/**
 * @author Marian Zlatev (mzlatev91@gmail.com)
 */
angular.module('category', ['gateway'])

        .controller('CategoryCtrl', ['$scope', 'categoryGateway', function ($scope, categoryGateway) {

          //$scope.categories = [];
          //
          //categoryGateway.fetchAll().then(
          //        function (data) {
          //          $scope.categories = data;
          //        });
          //
          //$scope.delete = function (category) {
          //
          //  categoryGateway.delete(category.id).then(
          //          function () {
          //            var index = $scope.categories.indexOf(category);
          //            $scope.categories.splice(index, 1);
          //          })
          //};

        }])

        .controller('CategoryRegistryCtrl', ['$scope', 'categoryGateway','$state', function ($scope, categoryGateway,$state) {

          //$scope.register = function (category) {
          //
          //  categoryGateway.register(category).then(
          //          function () {
          //            $state.go("/category");
          //          },
          //          function (message) {
          //            $scope.message = message;
          //          })
          //};
        }]);
